takwimu/views.py

```diff
@@ -174,50 +172,89 @@ class SearchView(TemplateView):
 
     def get(self, request, *args, **kwargs):
         search_query = request.GET.get('q', '')
+        self.country_filter = request.GET.getlist('country')
+        self.topic_filter = request.GET.getlist('topic')
+        self.topics_widgets_map = {}
+        self.create_results_map()
+
         self.items = []
         self.countries = OrderedDict()
         self.topics = OrderedDict()
-        if search_query:
-
-            # Profile page only indexes geo and body but not sections.
-            profilepage_results = ProfilePage.objects.live().search(
-                search_query)
-            for profilepage in profilepage_results.results():
-                self._extract_search_results(request, profilepage)
-
-            # Hence, sections (i.e. topics in the UI) need to be searched independent
-            # of profile page
-            profilesectionpage_results = ProfileSectionPage.objects.live().search(
-                search_query)
-            for profilesectionpage in profilesectionpage_results.results():
-                self._extract_search_results(request, profilesectionpage)
-
-            Query.get(search_query).add_hit()
+
+        takwimu_search = TakwimuTopicSearch()
+        # takwimu_search.es_index = 'test-topics-4'
+        # results = []
+        if search_query.startswith('"') and search_query.endswith('"'):
+            # search in quotes means phrase search
+            search_query = search_query.replace('"', '')
+            results = takwimu_search.search(search_query, operator="and", country_filters=self.country_filter, category_filters=self.topic_filter)
+        else:
+            results = takwimu_search.search(search_query, country_filters=self.country_filter, category_filters=self.topic_filter)
+
+        profilepages = ProfilePage.objects.live()
+        profilesectionpages = ProfileSectionPage.objects.live()
```

Comment:
You're loading these again in `create_results_map`, since they're in the same object and in the same method, you should load them only once.